src/MetaMorpho.sol

```diff
@@ -351,19 +351,22 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         uint256 totalWithdrawn;
         for (uint256 i; i < withdrawn.length; ++i) {
             MarketAllocation memory allocation = withdrawn[i];
+            Id id = allocation.marketParams.id();
```

Comment:
I'm not sure we should do this, because this is only used once in the code (once in each branch of the code)

Reply:
I think the code is cleaner like that, I would keep it